scripts/ws-session.sh

```
@@ -0,0 +1,100 @@
+# session id resolves.
+ws_session_identity_path() {
+    ws_session_identity_path_for "$(ws_resolve_session_id)"
+}
+
+# Echo the GDD_CO_AUTHOR value stored in an identity file, or empty if the
+# path is empty / missing / sets nothing. Sourced in a subshell so the file
+# can never leak into or corrupt the caller's environment.
+ws_read_identity_file() {
+    local path="${1:-}"
+    [[ -n "$path" && -f "$path" ]] || return 0
+    ( source "$path" >/dev/null 2>&1; printf '%s' "${GDD_CO_AUTHOR:-}" )
```

Shellcheck (0.11.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)